modules/rgbd/src/normal.cpp

```diff
@@ -129,13 +129,22 @@ namespace rgbd
   signNormal(const Vec<T, 3> & normal_in, Vec<T, 3> & normal_out)
   {
     Vec<T, 3> res;
-    if (normal_in[2] > 0)
-      res = -normal_in / norm_vec(normal_in);
+    if(norm_vec(normal_in) == 0) { //avoid division by 0
+      normal_out[0] = 0;
```

Comment:
not sure it's safer: we just need a convention. And NaN, well it's undefined.
It's like finding a convention for unknown depth: could be 0 in 16_U or NaN with float.